homeassistant/components/climate/comfoconnect.py

```diff
@@ -0,0 +1,275 @@
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/climate.comfoconnect/
+"""
+import logging
+import time
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.climate import (
+    ClimateDevice, PLATFORM_SCHEMA, ATTR_FAN_MODE, ATTR_CURRENT_HUMIDITY,
+    ATTR_CURRENT_TEMPERATURE, ATTR_FAN_LIST)
+from homeassistant.const import (
+    CONF_HOST, CONF_TOKEN, CONF_PIN, TEMP_CELSIUS, CONF_NAME)
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = [
+    'https://github.com/michaelarnauts/comfoconnect'
+    '/archive/0.1.zip#pycomfoconnect==0.1']
+
+SPEED_AWAY = 'away'
+SPEED_LOW = 'low'
+SPEED_MEDIUM = 'medium'
+SPEED_HIGH = 'high'
+
+ATTR_OUTSIDE_TEMPERATURE = 'outside_temperature'
+ATTR_OUTSIDE_HUMIDITY = 'outside_humidity'
+ATTR_AIR_FLOW_SUPPLY = 'air_flow_supply'
+ATTR_AIR_FLOW_EXTRACT = 'air_flow_extract'
+
+CONF_USER_AGENT = 'user_agent'
+
+DEFAULT_PIN = 0
+DEFAULT_TOKEN = '00000000000000000000000000000001'
+DEFAULT_NAME = 'ComfoAirQ'
+DEFAULT_USER_AGENT = 'Home Assistant'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_TOKEN, default=DEFAULT_TOKEN):
+        vol.Length(min=32, max=32, msg='invalid token'),
+    vol.Optional(CONF_USER_AGENT, default=DEFAULT_USER_AGENT):
+        cv.string,
+    vol.Optional(CONF_PIN, default=DEFAULT_PIN): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ComfoConnect bridge."""
+    from pycomfoconnect import (Bridge)
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+    user_agent = config.get(CONF_USER_AGENT)
+    pin = config.get(CONF_PIN)
+
+    # Run discovery on the configured ip
+    bridges = Bridge.discover(host)
+    if not bridges:
+        _LOGGER.error('Could not connect to ComfoConnect bridge on %s', host)
+        return False
+
+    for bridge in bridges:
+        _LOGGER.info('Bridge found: %s (%s)', bridge.remote_uuid.hex(),
+                     bridge.ip)
+        add_devices([
+            ComfoConnectBridge(hass, name, bridge, token, user_agent, pin)
+        ], True)
+
+    return
+
+
+class ComfoConnectBridge(ClimateDevice):
+    """Representation of a ComfoConnect bridge."""
+
+    def __init__(self, hass, name, bridge, token, friendly_name, pin):
+        """Initialize the ComfoConnect bridge."""
+        from pycomfoconnect import (
+            ComfoConnect, SENSOR_FAN_SPEED_MODE, SENSOR_TEMPERATURE_EXTRACT,
+            SENSOR_TEMPERATURE_OUTDOOR, SENSOR_HUMIDITY_EXTRACT,
+            SENSOR_HUMIDITY_OUTDOOR, SENSOR_FAN_SUPPLY_FLOW,
+            SENSOR_FAN_EXHAUST_FLOW
+        )
+
+        self._hass = hass
+        self._name = name
+        self._comfoconnect = ComfoConnect(bridge, self.sensor_callback)
```

Comment:
The callback thread of comfoconnect is only started when the `update` is executed (when `self._comfoconnect.connect` is executed). It is stopped in `self._comfoconnect.disconnect` in the finally clause. It only stays awake for maximum 5 seconds (during `wait_for_sensor_values`).

I initially used the EVENT_HOMEASSISTANT_[START|STOP] event, and kept the callback thread open, but the bridge only allows one open connection at the same time, so I decided to stay with polling and open the connection during `update`.